src/components/Button/Button.styles.ts

```diff
@@ -0,0 +1,93 @@
+/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
+import { palette } from '../../util/colors';
+import { IButtonStyles } from 'office-ui-fabric-react/lib/components/Button/Button.types';
+import { ButtonColor, ButtonSize, ButtonStatus } from './enums';
+import { ButtonProps } from './button';
+import Styles from '../../util/styles';
+
+const smallPadding = '4px 9px 6px';
+const regularPadding = '7px 15px';
+
+const primaryColors: IButtonStyles = {
+  root: {
+    color: palette.white,
+    backgroundColor: palette.themeDark,
+    borderColor: palette.themeDark,
+  },
+  rootHovered: {
+    color: palette.white,
+    backgroundColor: palette.themeDarkAlt,
+    borderColor: palette.themeDarkAlt,
+    textDecoration: 'none',
+  },
+  rootFocused: {
+    color: palette.white,
+    backgroundColor: palette.themeDarkAlt,
+    borderColor: palette.themeDarkAlt,
+  },
+  rootPressed: {
+    color: palette.white,
+    backgroundColor: palette.themeDarker,
+    borderColor: palette.themeDarker,
+  },
+};
+
+const secondaryColors: IButtonStyles = {
+  root: {
+    color: palette.themeDark,
+    backgroundColor: palette.white,
+    borderColor: palette.neutralDark,
+  },
+  rootHovered: {
+    color: palette.themeDark,
+    backgroundColor: palette.white,
+    borderColor: palette.themeLighter,
+    textDecoration: 'none',
+  },
+  rootFocused: {
+    color: palette.themeDark,
+    backgroundColor: palette.white,
+    borderColor: palette.themeLighter,
+  },
+  rootPressed: {
+    color: palette.themeDark,
+    backgroundColor: palette.white,
+    borderColor: palette.themePrimary,
+  },
+};
+
+export const getStyles = (props: ButtonProps): IButtonStyles => {
+  const { size, color, status, fullWidth } = props;
+
+  const colors = color === ButtonColor.PRIMARY ? primaryColors : secondaryColors;
+  const { root, rootHovered, rootFocused, rootPressed } = colors;
+
+  return {
+    root: {
+      ...(root as {}),
```

Comment:
why is the `as {}` necessary?

Reply:
`[ts] Spread types may only be created from object types.`

Reply:
`root` is an object, i was just wondering why the `as {}` was necessary but not a big deal